scripts/api_e2e.py

```diff
@@ -0,0 +1,122 @@
+
+
+def main(args):
+
+    api_url = f"http://localhost:{args.port}"
+
+    # Log as superuser
+    superuser_login = getpass('Login: ') if args.creds else "superuser"
+    superuser_pwd = getpass() if args.creds else "superuser"
+
+    # Retrieve superuser token
+    superuser_auth = {
+        "Authorization": f"Bearer {get_token(api_url, superuser_login, superuser_pwd)}",
+        "Content-Type": "application/json",
+    }
+
+    user_login = 'my_user'
+    user_pwd = 'my_pwd'
+
+    # create a user
+    payload = dict(login=user_login, password=user_pwd, scopes="me")
+    user_id = api_request('post', f"{api_url}/users/", payload, superuser_auth)['id']
+    user_auth = {
+        "Authorization": f"Bearer {get_token(api_url, user_login, user_pwd)}",
+        "Content-Type": "application/json",
+    }
+
+    # Create a site
+    payload = dict(name='first_site', country="FR", geocode="01", lat=44.1, lon=3.9)
+    site_id = api_request('post', f"{api_url}/sites/", payload, superuser_auth)['id']
+
+    # Update the user password
+    payload = dict(password='my_second_pwd')
+    api_request('put', f"{api_url}/users/update-pwd", payload, user_auth)
+
+    # Create a device (as admin until #79 is closed)
+    device_login = 'my_device'
+    device_pwd = 'my_third_password'
+    payload = dict(login=device_login, password=device_pwd, specs="raspberry_pi")
+    device_id = api_request('post', f"{api_url}/devices/register", payload, user_auth)['id']
+
+    device_auth = {
+        "Authorization": f"Bearer {get_token(api_url, device_login, device_pwd)}",
+        "Content-Type": "application/json",
+    }
+
+    # create an installation with this device and the site
+    payload = dict(device_id=device_id, site_id=site_id, lat=44.1, lon=3.9, elevation=100., yaw=0., pitch=0.,
+                   start_ts="2019-08-24T14:15:22.00")
+    installation_id = api_request('post', f"{api_url}/installations/", payload, superuser_auth)['id']
+
+    # Installation creates a media
+    payload = dict(type='image')
+    media_id = api_request('post', f"{api_url}/media/from-device", payload, device_auth)['id']
+
+    # Installation creates an event & alert
+    payload = dict(lat=44.1, lon=3.9, type='wildfire')
+    event_id = api_request('post', f"{api_url}/events/", payload, superuser_auth)['id']
+
+    payload = dict(lat=44.1, lon=3.9, event_id=event_id, media_id=media_id, type='start')
+    alert_id1 = api_request('post', f"{api_url}/alerts/from-device", payload, device_auth)['id']
+
+    # Installation throws the end alert
+    payload = dict(lat=44.1, lon=3.9, event_id=event_id, type='end')
+    alert_id2 = api_request('post', f"{api_url}/alerts/from-device", payload, device_auth)['id']
+
+    # Cleaning (order is important because of foreign key protection in existing tables)
+    response = requests.delete(f"{api_url}/alerts/{alert_id1}/", headers=superuser_auth)
+    assert response.status_code // 100 == 2, print(response.json()['detail'])
```

Comment:
can't you use the api_request method above? you are already doing the assert checks in it.

Reply:
We don't pass any payload with delete methods, but I can perhaps change `api_request`